.github/workflows/bundle-size-analysis.yml

```diff
@@ -0,0 +1,109 @@
+name: "Bundle size"
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:  
+  # Build main branch and upload its's bundle stats file
+  build-main:
+    name: "Building master"
+    runs-on: ubuntu-latest
+  
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout master branch 
+        uses: actions/checkout@v3.0.0
+        with:
+          ref: master
+
+      - name: Setting up Node v16.10.0
+        uses: actions/setup-node@v3.4.1
+        with:
+          node-version: 16.10.0
+          
+      - name: Install dependencies
+        uses: bahmutov/npm-install@v1.6.0

```

Comment:
using this as it caches the dependencies and restore if available, speeds up next installation run